This should be fixed in 3.9, but is unlikely to be backported.

--- linux-3.8.1-201.fc18.x86_64/drivers/tty/tty_buffer.c~	2013-03-01 11:07:37.498291384 -0500
+++ linux-3.8.1-201.fc18.x86_64/drivers/tty/tty_buffer.c	2013-03-01 11:08:11.088250537 -0500
@@ -473,7 +473,7 @@ static void flush_to_ldisc(struct work_s
 	struct tty_ldisc *disc;
 
 	tty = port->itty;
-	if (WARN_RATELIMIT(tty == NULL, "tty is NULL\n"))
+	if (tty == NULL)
 		return;
 
 	disc = tty_ldisc_ref(tty);
